src/terminal/input/terminalInput.cpp

```diff
@@ -706,60 +662,58 @@ bool TerminalInput::HandleKey(const IInputEvent* const pInEvent)
     // These mappings will kick in no matter which modifiers are pressed and as such
     // must be checked last, or otherwise we'd override more complex key combinations.
     const auto mapping = _getKeyMapping(keyEvent, _inputMode.test(Mode::Ansi), _inputMode.test(Mode::CursorKey), _inputMode.test(Mode::Keypad));
-    if (_translateDefaultMapping(keyEvent, mapping, senderFunc))
+    if (const auto match = _searchKeyMapping(keyEvent, mapping))
     {
-        return true;
+        return MakeOutput(match->sequence);
     }
 
     // If all else fails we can finally try to send the character itself if there is any.
     if (keyEvent.GetCharData() != 0)
     {
-        _SendChar(keyEvent.GetCharData());
-        return true;
+        return _SendChar(keyEvent.GetCharData());
     }
 
-    return false;
+    return MakeUnhandled();
 }
 
-bool TerminalInput::HandleFocus(const bool focused) noexcept
+TerminalInput::OutputType TerminalInput::HandleFocus(const bool focused) const
 {
-    const auto enabled{ _inputMode.test(Mode::FocusEvent) };
-    if (enabled)
+    if (!_inputMode.test(Mode::FocusEvent))
     {
-        _SendInputSequence(focused ? L"\x1b[I" : L"\x1b[O");
+        return MakeUnhandled();
     }
-    return enabled;
+
+    return MakeOutput(focused ? L"\x1b[I" : L"\x1b[O");
 }
 
 // Routine Description:
 // - Sends the given character to the shell.
 // - Surrogate pairs are being aggregated by this function before being sent.
 // Arguments:
 // - ch: The UTF-16 character to send.
-void TerminalInput::_SendChar(const wchar_t ch)
+TerminalInput::OutputType TerminalInput::_SendChar(const wchar_t ch)
 {
+    StringType str;
+
     if (til::is_leading_surrogate(ch))
     {
         if (_leadingSurrogate.has_value())
         {
-            // we already were storing a leading surrogate but we got another one. Go ahead and send the
-            // saved surrogate piece and save the new one
-            const auto formatted = wil::str_printf<std::wstring>(L"%I32u", _leadingSurrogate.value());
-            _SendInputSequence(formatted);
+            str.push_back(_leadingSurrogate.value());
         }
-        // save the leading portion of a surrogate pair so that they can be sent at the same time

```

Comment:
didn't want to keep this comment? it still applies!

Reply:
I didn't feel like it'd be a useful comment. Let me rephrase it to be a bit shorter.